### PR TITLE
Update nova policy for nova 2.1 API

### PR DESCRIPTION
**Pull request by j2sol**
This syncs our policy changes for API v2 with API v2.1. Many extensions
became part of the 2.1 API itself, so there are duplicate rules to deal
with.

Also:
  - restrict migration to full admin (but allow others to see
    migrations)

Change-Id: I4b4c1fbfa160a4ecf715ee03aa53c0470a933d2a

### CODE REVIEW

**Comment by j2sol** [comment]
Current diff against upstream:
```
--- policy.json.upstream	2016-06-08 21:31:06.000000000 -0700
+++ policy.json	2016-06-08 21:56:48.000000000 -0700
@@ -2,6 +2,17 @@
     "context_is_admin":  "role:admin",
     "admin_or_owner":  "is_admin:True or project_id:%(project_id)s",
     "default": "rule:admin_or_owner",
+    "is_projAdmin": "role:project_admin and project_id:%(project_id)s",
+    "is_cloudAdmin": "role:cloud_admin",
+    "member_role": "role:_member_",
+    "projAdmin_or_cloudAdmin": "rule:is_projAdmin or rule:is_cloudAdmin",
+    "is_resource_owner": "project_id:%(project_id)s and user_id:%(user_id)s",
+    "is_owner": "project_id:%(project_id)s",
+    "admin_or_projAdmin" :"role:admin or rule:is_projAdmin",
+    "admin_or_projAdmin_or_cloudAdmin": "role:admin or rule:projAdmin_or_cloudAdmin",
+    "admin_or_projAdmin_or_cloudAdmin_or_owner": "rule:admin_or_projAdmin_or_cloudAdmin or rule:is_resource_owner",
+    "admin_or_projAdmin_or_cloudAdmin_or_member": "rule:admin_or_projAdmin_or_cloudAdmin or rule:member_role",
+    "admin_or_cloudadmin": "role:admin or role:cloud_admin",

     "cells_scheduler_filter:TargetCellFilter": "is_admin:True",

@@ -12,7 +23,7 @@

     "compute:get": "rule:admin_or_owner",
     "compute:get_all": "rule:admin_or_owner",
-    "compute:get_all_tenants": "is_admin:True",
+    "compute:get_all_tenants": "rule:admin_or_cloudadmin",

     "compute:update": "rule:admin_or_owner",

@@ -25,12 +36,12 @@
     "compute:get_diagnostics": "rule:admin_or_owner",
     "compute:get_instance_diagnostics": "rule:admin_or_owner",

-    "compute:start": "rule:admin_or_owner",
-    "compute:stop": "rule:admin_or_owner",
+    "compute:start": "rule:admin_or_projAdmin_or_cloudAdmin_or_owner",
+    "compute:stop": "rule:admin_or_projAdmin_or_cloudAdmin_or_owner",

     "compute:lock": "rule:admin_or_owner",
     "compute:unlock": "rule:admin_or_owner",
-    "compute:unlock_override": "rule:admin_api",
+    "compute:unlock_override": "rule:admin_or_projAdmin_or_cloudAdmin",

     "compute:get_vnc_console": "rule:admin_or_owner",
     "compute:get_spice_console": "rule:admin_or_owner",
@@ -90,18 +101,18 @@

     "admin_api": "is_admin:True",
     "compute_extension:accounts": "rule:admin_api",
-    "compute_extension:admin_actions": "rule:admin_api",
-    "compute_extension:admin_actions:pause": "rule:admin_or_owner",
-    "compute_extension:admin_actions:unpause": "rule:admin_or_owner",
-    "compute_extension:admin_actions:suspend": "rule:admin_or_owner",
-    "compute_extension:admin_actions:resume": "rule:admin_or_owner",
-    "compute_extension:admin_actions:lock": "rule:admin_or_owner",
-    "compute_extension:admin_actions:unlock": "rule:admin_or_owner",
+    "compute_extension:admin_actions": "rule:admin_or_projAdmin_or_cloudAdmin",
+    "compute_extension:admin_actions:pause": "rule:admin_or_projAdmin_or_cloudAdmin_or_owner",
+    "compute_extension:admin_actions:unpause": "rule:admin_or_projAdmin_or_cloudAdmin_or_owner",
+    "compute_extension:admin_actions:suspend": "rule:admin_or_projAdmin_or_cloudAdmin_or_owner",
+    "compute_extension:admin_actions:resume": "rule:admin_or_projAdmin_or_cloudAdmin_or_owner",
+    "compute_extension:admin_actions:lock": "rule:admin_or_projAdmin_or_cloudAdmin_or_owner",
+    "compute_extension:admin_actions:unlock": "rule:admin_or_projAdmin_or_cloudAdmin_or_owner",
     "compute_extension:admin_actions:resetNetwork": "rule:admin_api",
-    "compute_extension:admin_actions:injectNetworkInfo": "rule:admin_api",
-    "compute_extension:admin_actions:createBackup": "rule:admin_or_owner",
+    "compute_extension:admin_actions:injectNetworkInfo": "rule:admin_or_projAdmin_or_cloudAdmin",
+    "compute_extension:admin_actions:createBackup": "rule:admin_or_projAdmin_or_cloudAdmin_or_owner",
     "compute_extension:admin_actions:migrateLive": "rule:admin_api",
-    "compute_extension:admin_actions:resetState": "rule:admin_api",
+    "compute_extension:admin_actions:resetState": "rule:admin_or_projAdmin_or_cloudAdmin",
     "compute_extension:admin_actions:migrate": "rule:admin_api",
     "compute_extension:aggregates": "rule:admin_api",
     "compute_extension:agents": "rule:admin_api",
@@ -122,27 +133,27 @@
     "compute_extension:deferred_delete": "rule:admin_or_owner",
     "compute_extension:disk_config": "rule:admin_or_owner",
     "compute_extension:evacuate": "rule:admin_api",
-    "compute_extension:extended_server_attributes": "rule:admin_api",
+    "compute_extension:extended_server_attributes": "rule:admin_or_projAdmin_or_cloudAdmin",
     "compute_extension:extended_status": "rule:admin_or_owner",
     "compute_extension:extended_availability_zone": "rule:admin_or_owner",
     "compute_extension:extended_ips": "rule:admin_or_owner",
     "compute_extension:extended_ips_mac": "rule:admin_or_owner",
     "compute_extension:extended_vif_net": "rule:admin_or_owner",
     "compute_extension:extended_volumes": "rule:admin_or_owner",
-    "compute_extension:fixed_ips": "rule:admin_api",
+    "compute_extension:fixed_ips": "rule:admin_or_projAdmin_or_cloudAdmin",
     "compute_extension:flavor_access": "rule:admin_or_owner",
-    "compute_extension:flavor_access:addTenantAccess": "rule:admin_api",
-    "compute_extension:flavor_access:removeTenantAccess": "rule:admin_api",
+    "compute_extension:flavor_access:addTenantAccess": "rule:admin_or_cloudadmin",
+    "compute_extension:flavor_access:removeTenantAccess": "rule:admin_or_cloudadmin",
     "compute_extension:flavor_disabled": "rule:admin_or_owner",
     "compute_extension:flavor_rxtx": "rule:admin_or_owner",
     "compute_extension:flavor_swap": "rule:admin_or_owner",
     "compute_extension:flavorextradata": "rule:admin_or_owner",
     "compute_extension:flavorextraspecs:index": "rule:admin_or_owner",
     "compute_extension:flavorextraspecs:show": "rule:admin_or_owner",
-    "compute_extension:flavorextraspecs:create": "rule:admin_api",
-    "compute_extension:flavorextraspecs:update": "rule:admin_api",
-    "compute_extension:flavorextraspecs:delete": "rule:admin_api",
-    "compute_extension:flavormanage": "rule:admin_api",
+    "compute_extension:flavorextraspecs:create": "rule:admin_or_cloudadmin",
+    "compute_extension:flavorextraspecs:update": "rule:admin_or_cloudadmin",
+    "compute_extension:flavorextraspecs:delete": "rule:admin_or_cloudadmin",
+    "compute_extension:flavormanage": "rule:admin_or_cloudadmin",
     "compute_extension:floating_ip_dns": "rule:admin_or_owner",
     "compute_extension:floating_ip_pools": "rule:admin_or_owner",
     "compute_extension:floating_ips": "rule:admin_or_owner",
@@ -151,7 +162,7 @@
     "compute_extension:fping:all_tenants": "rule:admin_api",
     "compute_extension:hide_server_addresses": "is_admin:False",
     "compute_extension:hosts": "rule:admin_api",
-    "compute_extension:hypervisors": "rule:admin_api",
+    "compute_extension:hypervisors": "rule:admin_or_cloudAdmin",
     "compute_extension:image_size": "rule:admin_or_owner",
     "compute_extension:instance_actions": "rule:admin_or_owner",
     "compute_extension:instance_actions:events": "rule:admin_api",
@@ -166,10 +177,10 @@
     "compute_extension:networks:view": "rule:admin_or_owner",
     "compute_extension:networks_associate": "rule:admin_api",
     "compute_extension:os-tenant-networks": "rule:admin_or_owner",
-    "compute_extension:quotas:show": "rule:admin_or_owner",
-    "compute_extension:quotas:update": "rule:admin_api",
-    "compute_extension:quotas:delete": "rule:admin_api",
-    "compute_extension:quota_classes": "rule:admin_or_owner",
+    "compute_extension:quotas:show": "rule:admin_or_projAdmin_or_cloudAdmin or rule:is_owner",
+    "compute_extension:quotas:update": "rule:admin_or_cloudadmin",
+    "compute_extension:quotas:delete": "rule:admin_or_cloudadmin",
+    "compute_extension:quota_classes": "rule:admin_or_projAdmin_or_cloudAdmin or rule:is_owner",
     "compute_extension:rescue": "rule:admin_or_owner",
     "compute_extension:security_group_default_rules": "rule:admin_api",
     "compute_extension:security_groups": "rule:admin_or_owner",
@@ -179,9 +190,9 @@
     "compute_extension:server_usage": "rule:admin_or_owner",
     "compute_extension:services": "rule:admin_api",
     "compute_extension:shelve": "rule:admin_or_owner",
-    "compute_extension:shelveOffload": "rule:admin_api",
-    "compute_extension:simple_tenant_usage:show": "rule:admin_or_owner",
-    "compute_extension:simple_tenant_usage:list": "rule:admin_api",
+    "compute_extension:shelveOffload": "rule:admin_or_projAdmin_or_cloudAdmin",
+    "compute_extension:simple_tenant_usage:show": "rule:admin_or_projAdmin_or_cloudAdmin or rule:is_owner",
+    "compute_extension:simple_tenant_usage:list": "rule:admin_or_projAdmin_or_cloudAdmin",
     "compute_extension:unshelve": "rule:admin_or_owner",
     "compute_extension:users": "rule:admin_api",
     "compute_extension:virtual_interfaces": "rule:admin_or_owner",
@@ -194,9 +205,9 @@
     "compute_extension:volume_attachments:delete": "rule:admin_or_owner",
     "compute_extension:volumetypes": "rule:admin_or_owner",
     "compute_extension:availability_zone:list": "rule:admin_or_owner",
-    "compute_extension:availability_zone:detail": "rule:admin_api",
+    "compute_extension:availability_zone:detail": "rule:admin_or_projAdmin_or_cloudAdmin",
     "compute_extension:used_limits_for_admin": "rule:admin_api",
-    "compute_extension:migrations:index": "rule:admin_api",
+    "compute_extension:migrations:index": "rule:admin_or_projAdmin_or_cloudAdmin",
     "compute_extension:os-assisted-volume-snapshots:create": "rule:admin_api",
     "compute_extension:os-assisted-volume-snapshots:delete": "rule:admin_api",
     "compute_extension:console_auth_tokens": "rule:admin_api",
@@ -248,8 +259,8 @@
     "network:attach_external_network": "rule:admin_api",
     "network:get_vif_by_mac_address": "rule:admin_or_owner",

-    "os_compute_api:servers:detail:get_all_tenants": "is_admin:True",
-    "os_compute_api:servers:index:get_all_tenants": "is_admin:True",
+    "os_compute_api:servers:detail:get_all_tenants": "rule:admin_or_cloudadmin",
+    "os_compute_api:servers:index:get_all_tenants": "rule:admin_or_cloudadmin",
     "os_compute_api:servers:confirm_resize": "rule:admin_or_owner",
     "os_compute_api:servers:create": "rule:admin_or_owner",
     "os_compute_api:servers:create:attach_network": "rule:admin_or_owner",
@@ -273,14 +284,14 @@
     "os_compute_api:servers:migrations:force_complete": "rule:admin_api",
     "os_compute_api:servers:migrations:delete": "rule:admin_api",
     "os_compute_api:servers:discoverable": "@",
-    "os_compute_api:servers:migrations:index": "rule:admin_api",
-    "os_compute_api:servers:migrations:show": "rule:admin_api",
+    "os_compute_api:servers:migrations:index": "rule:admin_or_projAdmin_or_cloudAdmin_or_owner",
+    "os_compute_api:servers:migrations:show": "rule:admin_or_projAdmin_or_cloudAdmin_or_owner",
     "os_compute_api:os-access-ips:discoverable": "@",
     "os_compute_api:os-access-ips": "rule:admin_or_owner",
     "os_compute_api:os-admin-actions": "rule:admin_api",
     "os_compute_api:os-admin-actions:discoverable": "@",
     "os_compute_api:os-admin-actions:reset_network": "rule:admin_api",
-    "os_compute_api:os-admin-actions:inject_network_info": "rule:admin_api",
+    "os_compute_api:os-admin-actions:inject_network_info": "rule:admin_or_projAdmin_or_cloudAdmin_or_owner",
     "os_compute_api:os-admin-actions:reset_state": "rule:admin_api",
     "os_compute_api:os-admin-password": "rule:admin_or_owner",
     "os_compute_api:os-admin-password:discoverable": "@",
@@ -375,7 +386,7 @@
     "os_compute_api:os-hide-server-addresses:discoverable": "@",
     "os_compute_api:os-hosts": "rule:admin_api",
     "os_compute_api:os-hosts:discoverable": "@",
-    "os_compute_api:os-hypervisors": "rule:admin_api",
+    "os_compute_api:os-hypervisors": "rule:admin_or_cloudAdmin",
     "os_compute_api:os-hypervisors:discoverable": "@",
     "os_compute_api:images:discoverable": "@",
     "os_compute_api:image-size": "rule:admin_or_owner",
@@ -397,8 +408,8 @@
     "os_compute_api:limits:discoverable": "@",
     "os_compute_api:limits": "rule:admin_or_owner",
     "os_compute_api:os-lock-server:discoverable": "@",
-    "os_compute_api:os-lock-server:lock": "rule:admin_or_owner",
-    "os_compute_api:os-lock-server:unlock": "rule:admin_or_owner",
+    "os_compute_api:os-lock-server:lock": "rule:admin_or_projAdmin_or_cloudAdmin_or_owner",
+    "os_compute_api:os-lock-server:unlock": "rule:admin_or_projAdmin_or_cloudAdmin_or_owner",
     "os_compute_api:os-lock-server:unlock:unlock_override": "rule:admin_api",
     "os_compute_api:os-migrate-server:discoverable": "@",
     "os_compute_api:os-migrate-server:migrate": "rule:admin_api",
@@ -411,8 +422,8 @@
     "os_compute_api:os-networks-associate": "rule:admin_api",
     "os_compute_api:os-networks-associate:discoverable": "@",
     "os_compute_api:os-pause-server:discoverable": "@",
-    "os_compute_api:os-pause-server:pause": "rule:admin_or_owner",
-    "os_compute_api:os-pause-server:unpause": "rule:admin_or_owner",
+    "os_compute_api:os-pause-server:pause": "rule:admin_or_projAdmin_or_cloudAdmin_or_owner",
+    "os_compute_api:os-pause-server:unpause": "rule:admin_or_projAdmin_or_cloudAdmin_or_owner",
     "os_compute_api:os-pci:pci_servers": "rule:admin_or_owner",
     "os_compute_api:os-pci:discoverable": "@",
     "os_compute_api:os-pci:index": "rule:admin_api",
@@ -421,13 +432,13 @@
     "os_compute_api:os-personality:discoverable": "@",
     "os_compute_api:os-preserve-ephemeral-rebuild:discoverable": "@",
     "os_compute_api:os-quota-sets:discoverable": "@",
-    "os_compute_api:os-quota-sets:show": "rule:admin_or_owner",
+    "os_compute_api:os-quota-sets:show": "rule:admin_or_projAdmin_or_cloudAdmin or rule:is_owner",
     "os_compute_api:os-quota-sets:defaults": "@",
-    "os_compute_api:os-quota-sets:update": "rule:admin_api",
-    "os_compute_api:os-quota-sets:delete": "rule:admin_api",
-    "os_compute_api:os-quota-sets:detail": "rule:admin_api",
-    "os_compute_api:os-quota-class-sets:update": "rule:admin_api",
-    "os_compute_api:os-quota-class-sets:show": "is_admin:True or quota_class:%(quota_class)s",
+    "os_compute_api:os-quota-sets:update": "rule:admin_or_cloudadmin",
+    "os_compute_api:os-quota-sets:delete": "rule:admin_or_cloudadmin",
+    "os_compute_api:os-quota-sets:detail": "rule:admin_or_cloudadmin",
+    "os_compute_api:os-quota-class-sets:update": "rule:admin_or_cloudadmin",
+    "os_compute_api:os-quota-class-sets:show": "rule:admin_or_cloudadmin or quota_class:%(quota_class)s",
     "os_compute_api:os-quota-class-sets:discoverable": "@",
     "os_compute_api:os-rescue": "rule:admin_or_owner",
     "os_compute_api:os-rescue:discoverable": "@",
@@ -460,8 +471,8 @@
     "os_compute_api:os-simple-tenant-usage:show": "rule:admin_or_owner",
     "os_compute_api:os-simple-tenant-usage:list": "rule:admin_api",
     "os_compute_api:os-suspend-server:discoverable": "@",
-    "os_compute_api:os-suspend-server:suspend": "rule:admin_or_owner",
-    "os_compute_api:os-suspend-server:resume": "rule:admin_or_owner",
+    "os_compute_api:os-suspend-server:suspend": "rule:admin_or_projAdmin_or_cloudAdmin_or_owner",
+    "os_compute_api:os-suspend-server:resume": "rule:admin_or_projAdmin_or_cloudAdmin_or_owner",
     "os_compute_api:os-tenant-networks": "rule:admin_or_owner",
     "os_compute_api:os-tenant-networks:discoverable": "@",
     "os_compute_api:os-shelve:unshelve": "rule:admin_or_owner",
```

File: roles/nova-common/templates/etc/nova/policy.json [hunk 1]
```diff
@@ -111,9 +111,9 @@
     "compute_extension:admin_actions:resetNetwork": "rule:admin_api",
     "compute_extension:admin_actions:injectNetworkInfo": "rule:admin_or_projAdmin_or_cloudAdmin",
     "compute_extension:admin_actions:createBackup": "rule:admin_or_projAdmin_or_cloudAdmin_or_owner",
-    "compute_extension:admin_actions:migrateLive": "rule:admin_or_projAdmin_or_cloudAdmin",
+    "compute_extension:admin_actions:migrateLive": "rule:admin_api",
```
Comment (zfeldstein) [inline roles/nova-common/templates/etc/nova/policy.json]: How come we don't want cloud_admin to have this permission?

Comment (j2sol) [inline roles/nova-common/templates/etc/nova/policy.json]: Live migration is a thing we are exposing to customers. It's not reliable, it doesn't honor scheduling constraints, and it's not a thing we want our users to depend on. We've only implemented it to facilitate host evacuations in the case of disruptive maintenance, and those are only to be initiated by IBM admins.

**Comment by zfeldstein** [comment]
LGTM 👍

**Pull request merged**